enumerables.rb

```diff
@@ -0,0 +1,154 @@
+module Enumerable
+  def my_each
+    each do |value|
+      yield(value)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    each do |value|
+      yield(value, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    each do |value|
+      new_array.push(value) if yield(value)
+    end
+    new_array
+  end
+
+  def my_all?
+    flag = true
+    each do |value|
+      flag = yield(value)
+      break unless flag
+    end
+    flag
+  end
+
+  def my_any?
+    flag = false
+    each do |value|
+      flag = yield(value)
+      break if flag
+    end
+    flag
+  end
+
+  def my_none?
+    flag = false
+    each do |value|
+      flag = yield(value)
+      break if flag
+    end
+    !flag
+  end
+
+  def my_count
+    i = 0
+    if block_given?
+      each do |value|
+        i += 1 if yield(value)
+      end
+    else
+      i = length
+    end
+    i
+  end
+
+  def my_map
+    new_array = []
+    each do |value|
+      new_array.push(yield(value))
+    end
+    new_array
+  end
+
+  def my_inject
+    counter = 0
+    if block_given?
+      each do |value|
+        counter += value if yield(value)
+      end
+    else
+      each do |value|
+        counter += value
+      end
+    end
+    counter
+  end
```

Comment:
* [ ] `my_inject` is identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_inject combines all elements of enum by applying a binary operation, specified by a block
  ```ruby
    array = [4,7,8,89,9,6,5,12,4,2,5]
    operation = proc { |sum, n| sum + n  }
    p array.my_inject(&operation)  # returns this error
    # TypeError:
        nil can't be coerced into Integer
      # ./spec/enumerables_spec.rb:206:in `+'
      # ./spec/enumerables_spec.rb:206:in `block (4 levels) in <top (required)>'
      # ./enumerables.rb:75:in `block in my_inject'
      # ./enumerables.rb:74:in `each'
      # ./enumerables.rb:74:in `my_inject'
  ```
     

* [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
  ```ruby
    array = [1,7,8,89,9,6,5,12,4,2,5]
    p array.my_inject(:+) # return this error
    # ArgumentError:
        wrong number of arguments (given 1, expected 0)
      # ./enumerables.rb:71:in `my_inject'
  ```